vignettes/data-preparation.Rmd

```diff
@@ -0,0 +1,399 @@
+---
+title: "Data Preparation and App Deployment"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Data Preparation and App Deployment}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  eval = FALSE
+)
+```
+
+```{r setup, include = FALSE}
+library(gsm.app)
+library(dplyr)
+library(purrr)
+```
+
+# Introduction
+
+The {gsm.app} package creates interactive dashboards for exploring Good Statistical Monitoring (GSM) Key Risk Indicator (KRI) assessments in clinical trials. This vignette provides a comprehensive guide on how to prepare data for use with the {gsm.app} Shiny application and how to deploy an app.
+
+The {gsm} suite of packages leverages Key Risk Indicators (KRIs) and thresholds to conduct study-level, country-level and site-level Risk Based Monitoring for clinical trials. The {gsm.app} uses {gsm.core} and {gsm.reporting} outputs to create all required data structures for the Shiny app and therefore this vignette will refer to the the functions and workflows that produce data frames, visualizations, metadata, and reports within those packages.
+
+The image below illustrates the overarching context in which the reporting workflow runs, taking inputs from both the output of the analytics workflow, as well as raw study-, site-, and country-level data in the Raw/Raw+ format.
+
+![](data_model_detailed.png){width="100%"}
+
+# Required Data Structures
```

Comment:
I... feel like all of this has too much detail. We should point to the other gsm documentation about preparing data. The unique requirements that we add are `fnFetchData()` and (optionally) `fnCountData()`. Everything from here through line 304 should be condensed to a brief description and a link to probably {gsm.core} documentation.